lib/micromachine.rb

```diff
@@ -44,9 +49,21 @@ def states
 private
 
   def change(event)
-    @state = transitions_for[event][@state]
+    previous_state, @state = @state, transitions_for[event][@state]
+
+    all_arguments = [event, previous_state, state]
```

Comment:
Nope. 

There interface to access the state: `state` method. Direct reads from the internal state representation are breaking encapsulation. Should the `@state` representation change, for example, to hold integer values internally, it will break code that reads ivars directly.

Speaking from the SOLID point of view, there too much responsibilities on the `change` method. Because not it only changes the state, but also notifies callbacks. Ideally, callback notification code should be extracted to the, um, `notify` method.